File: lookups/v1/phone_number/fetch-lookup-get-addon-payfone-example-1/fetch-lookup-get-addon-payfone-example-1.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using System.Collections.Generic;
using Twilio;
using Twilio.Rest.Lookups.V1;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var addOns = new List<string> {
            "payfone_tcpa_compliance"
        };

        var addOnsData = new Dictionary<string, Object>()
        {
            {"payfone_tcpa_compliance.right_party_contacted_date", 20160101}
        };

        var type = new List<string> {
            "carrier"
        };

        var phoneNumber = PhoneNumberResource.Fetch(
            addOns: addOns,
            addOnsData: addOnsData,
            type: type,
            pathPhoneNumber: new Twilio.Types.PhoneNumber("+15108675310")
        );

        Console.WriteLine(phoneNumber.CallerName);
    }
}

File: lookups/v1/phone_number/fetch-lookup-get-addon-payfone-example-1/fetch-lookup-get-addon-payfone-example-1.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

phone_number = @client.lookups
  .phone_numbers('+15108675310')
  .fetch(add_ons: 'payfone_tcpa_compliance', add_ons_data: {
       'payfone_tcpa_compliance.right_party_contacted_date' => 20160101
   }, type: 'carrier')

puts phone_number.caller_name
